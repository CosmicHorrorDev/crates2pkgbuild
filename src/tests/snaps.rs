use super::utils::{init_log, sample_package, FullFakeEnv, SampleCrate};

macro_rules! gen_snapshot_tests {
    ( $( ( $test_name:ident, $sample:expr ) ),* $(,)? ) => {
        $(
            #[test]
            fn $test_name() {
                init_log();

                let env = FullFakeEnv::new();
                let crate_ = sample_package($sample);
                let pkgbuild = crate::fetch_with_env(crate_.clone(), &env).unwrap();
                insta::with_settings!({ info => &crate_ }, {
                    insta::assert_snapshot!(pkgbuild);
                });
            }
        )*
    }
}

gen_snapshot_tests!(
    (cargo_pgo, SampleCrate::CargoPgo),
    (inlyne, SampleCrate::Inlyne),
    (ra_multiplex, SampleCrate::RaMultiplex),
    (regex_cli, SampleCrate::RegexCli),
    (to_html, SampleCrate::ToHtml),
);
