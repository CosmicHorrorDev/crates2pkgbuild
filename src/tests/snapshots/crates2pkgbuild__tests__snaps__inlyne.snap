---
source: src/tests/snaps.rs
expression: pkgbuild
info:
  name: inlyne
  val:
    maintainers:
      - Test <dummy@web.site>
    contributors: []
    version: 0.4.2
    release-version: 1
    arch:
      - x86_64
    deps:
      - fontconfig
      - freetype2
      - gcc-libs
      - libxcursor
      - libxi
      - libxrandr
      - oniguruma
    make-deps:
      - cargo
      - libffi
      - libxcb
      - libxkbcommon
      - wayland
    env:
      CARGO_TARGET_DIR:
        value: target
        steps:
          - build
          - check
      RUSTONIG_SYSTEM_LIBONIG:
        value: "yes"
        steps:
          - build
          - check
      RUSTUP_TOOLCHAIN:
        value: stable
        steps:
          - build
          - check
          - prepare
    test-in-check: true
    missing-license-files: false
    find-build-completions: []
    static-completions-within: completions
---
# Maintainer: Test <dummy@web.site>

pkgname=inlyne
pkgver=0.4.2
pkgrel=1
arch=(x86_64)
license=(MIT)
pkgdesc='Introducing Inlyne, a GPU powered yet browserless tool to help you quickly view markdown files in the blink of an eye.'
url='https://github.com/Inlyne-Project/inlyne'
depends=(fontconfig freetype2 gcc-libs libxcursor libxi libxrandr oniguruma)
makedepends=(cargo libffi libxcb libxkbcommon wayland)
source=("$pkgname-$pkgver.tar.gz::https://static.crates.io/crates/$pkgname/$pkgname-$pkgver.crate")
# Upstream checksum. Can be verified with
# $ curl https://index.crates.io/in/ly/inlyne | grep -F '"0.4.2"' | jq .cksum
sha256sums=(04df5c5066cb123252c4361d951e26f7a3b032c79f1e7d4db0924ffbc920ac24)

prepare() {
  cd "$srcdir/$pkgname-$pkgver"

  export RUSTUP_TOOLCHAIN=stable
  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
}

build() {
  cd "$srcdir/$pkgname-$pkgver"

  export CARGO_TARGET_DIR=target
  export RUSTONIG_SYSTEM_LIBONIG=yes
  export RUSTUP_TOOLCHAIN=stable
  cargo build --frozen --release
}

check() {
  cd "$srcdir/$pkgname-$pkgver"

  export CARGO_TARGET_DIR=target
  export RUSTONIG_SYSTEM_LIBONIG=yes
  export RUSTUP_TOOLCHAIN=stable
  cargo test --release --locked
}

package() {
  cd "$srcdir/$pkgname-$pkgver"

  install -Dm755 target/release/inlyne "$pkgdir/usr/bin/inlyne"
  install -Dm644 README.md "$pkgdir/usr/share/doc/$pkgname/README.md"
  install -Dm644 completions/_inlyne "$pkgdir/usr/share/zsh/site-functions/_inlyne"
  install -Dm644 completions/inlyne.elv "$pkgdir/usr/share/elvish/lib/inlyne.elv"
  install -Dm644 completions/inlyne.fish "$pkgdir/usr/share/fish/vendor_completions.d/inlyne.fish"
}

# vi: filetype=sh shiftwidth=2 expandtab
